bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1135 @@
+        try {
+            jsonElement = JsonParser.parseString(data);
+        } catch (JsonSyntaxException e) {
+            logger.debug("onEventData() invalid data '{}'", data, e);
+            return;
+        }
+        if (!(jsonElement instanceof JsonArray)) {
+            logger.debug("onEventData() data is not a JsonArray {}", data);
+            return;
+        }
+        List<Event> events;
+        try {
+            events = jsonParser.fromJson(jsonElement, Event.EVENT_LIST_TYPE);
+        } catch (JsonParseException e) {
+            logger.debug("onEventData() {}", e.getMessage(), e);
+            return;
+        }
+        if (Objects.isNull(events) || events.isEmpty()) {
+            logger.debug("onEventData() event list is null or empty");
+            return;
+        }
+        List<Resource> resources = new ArrayList<>();
+        events.forEach(event -> resources.addAll(event.getData()));
+        if (resources.isEmpty()) {
+            logger.debug("onEventData() resource list is empty");
+            return;
+        }
+        resources.forEach(resource -> resource.markAsSparse());
+        bridgeHandler.onResourcesEvent(resources);
+    }
+
+    /**
+     * Open the HTTP 2 session and the event stream.
+     *
+     * @throws ApiException if there was a communication error.
+     * @throws InterruptedException
+     */
+    public void open() throws ApiException, InterruptedException {
+        logger.debug("open()");
+        openPassive();
+        openActive();
+        bridgeHandler.onConnectionOnline();
+    }
+
+    /**
+     * Make the session active, by opening an HTTP 2 SSE event stream (if necessary).
+     *
+     * @throws ApiException if an error was encountered.
+     * @throws InterruptedException
+     */
+    private void openActive() throws ApiException, InterruptedException {
+        synchronized (this) {
+            openEventStream();
+            onlineState = State.ACTIVE;
+        }
+    }
+
+    /**
+     * Open the check alive task if necessary.
+     */
+    private void openCheckAliveTask() {
+        Future<?> task = checkAliveTask;
+        if (Objects.isNull(task) || task.isCancelled() || task.isDone()) {
+            logger.debug("openCheckAliveTask()");
+            cancelTask(checkAliveTask, false);
+            checkAliveTask = bridgeHandler.getScheduler().scheduleWithFixedDelay(() -> checkAlive(),
+                    CHECK_ALIVE_SECONDS, CHECK_ALIVE_SECONDS, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Implementation to open an HTTP 2 SSE event stream if necessary.
+     *
+     * @throws ApiException if an error was encountered.
+     * @throws InterruptedException
+     */
+    private void openEventStream() throws ApiException, InterruptedException {
+        Session session = http2Session;
+        if (Objects.isNull(session) || session.isClosed()) {
+            throw new ApiException("HTTP 2 session is null or in an illegal state");
```

Comment:
For same condition in `putResource`:
```suggestion
            throw new ApiException("HTTP 2 session is null or closed");
```

Reply:
Ok.